tasmota/tasmota_xsns_sensor/xsns_98_luxv30b.ino

```diff
@@ -0,0 +1,220 @@
+ *              1.44                      368.64           94003               25                 101           0   
+ *              1.44                      368.64           94003               200                010           1   
+ *              2.88                      737.28           200000              12.5               110           0   
+ *              2.88                      737.28           200000              100                011           1   
+ *              5.76                      737.28           200000              6.25               111           0   
+ *              5.76                      737.28           200000              50                 100           1   
+ *              11.52                     737.28           200000              25                 101           1   
+ *              23.04                     737.28           200000              12.5               110           1   
+ *              46.08                     737.28           200000              6.25               111           1   
+ *     —------------------------------------------------------------------------------------------------------------
+ * *************************************************************************************************/
+
+#define XSNS_98        98
+#define XI2C_69        69  // See I2CDEVICES.md
+
+#define LUXV30B_ADDR      0x4A  // Two wire library uses 7-bit addresses throughout
+#define LUXV30B_DATAREG   0x00   //the address of the data register
+#define LUXV30B_CONFREG   0x04   //the address of the configuration register
+
+class LUXV30B
+{
+  public:
+    LUXV30B();
+    void detect();
+    bool found();
+    void read();
+    float lux();
+  private:
+    bool _found;
+    uint32_t _lux;
+};
+
+LUXV30B::LUXV30B()
+{
+  _found = false;
+  _lux = 0;
+}
+
+float LUXV30B::lux()
+{
+  return ((float)_lux * 1.4) / 1000;
+}
+
+bool LUXV30B::found()
+{
+  return _found;
+}
+
+void LUXV30B::detect()
+{
+  if (I2cActive(LUXV30B_ADDR)) { return; }
+  Wire.beginTransmission(LUXV30B_ADDR);
+  if (0 == Wire.endTransmission()) {
+    _found = true;
+    I2cSetActiveFound(LUXV30B_ADDR, "LUXV30B");
+  }
+}
+
+void LUXV30B::read()
+{
+  _lux  = I2cRead8(LUXV30B_ADDR, 0);
+  delay(8);
+  _lux |= I2cRead8(LUXV30B_ADDR, 1) << 8;
+  delay(8);
+  _lux |= I2cRead8(LUXV30B_ADDR, 2) << 16;
+  delay(8);
+  _lux |= I2cRead8(LUXV30B_ADDR, 3) << 24;
+}
+
+LUXV30B luxv30b;
+
+/*********************************************************************************************\
+ * Interface
+\*********************************************************************************************/
+
+bool Xsns98(byte function)
+{
+  if (!I2cEnabled(XI2C_69))
+  {
+    return false;
+  }
+
+  bool result = false;
+
+  if (FUNC_INIT == function)
+  {
+    luxv30b.detect();
+  }
+
+  else if (luxv30b.found())
+  {
+    char lux[FLOATSZ];
+    dtostrfd(luxv30b.lux(), 2, lux);
+
```

Comment:
Pls consider the impact of this; on every function call this conversion code is run most of the time for noting.

Move the conversion to a new show function only to be executed at show time.

In addition I also think it's no use to measure this every second considering the delay of 24 mSecs per read.